lib/Newsletter/NewslettersRepository.php

```diff
@@ -126,6 +126,15 @@ public function bulkTrash(array $ids): int {
     if (empty($ids)) {
       return 0;
     }
+    // Fetch children id for trashing
+    $childrenIds = $this->entityManager->createQueryBuilder()->select( 'n.id')
+      ->from(NewsletterEntity::class, 'n')
+      ->where('n.parent IN (:ids)')
+      ->setParameter('ids', $ids)
+      ->getQuery()->getScalarResult();
```

Comment:
"Fetch children IDs for <action>" is always the same, right? Could be a private method.